### PR TITLE
Refactor lesson09. Mainly added React emphasis

### PR DESCRIPTION
**Pull request by yunhsincynthiachen**
Added a React tutorial and sample code, and updated lesson09 readme to have updated information about in-class activities, as well as combined all of lesson 9 readmes into one.

### CODE REVIEW

File: lessons/09-clientside-frameworks/README.md [hunk 1]
```diff
@@ -1,51 +1,57 @@
...
 
-You may wonder why this is not how it has always been done. The simple answer to this is that it is only recently that Javascript, browsers, and computers have gotten good enough to reliably utilize these client-side frameworks. If you were going to do anything interesting, you had to do it on the server. In addition, this method may be more labor intensive for certain projects and can be a tricky adjustment to make. You will get to explore some of these strengths and weaknesses in the next lab assignment when you tackle one of these frameworks yourself. 
+You may wonder why this is not how it has always been done. The simple answer to this is that it is only recently that Javascript, browsers, and computers have gotten good enough to reliably utilize these client-side frameworks. If you were going to do anything interesting, you had to do it on the server. In addition, this method may be more labor intensive for certain projects and can be a tricky adjustment to make. You will get to explore some of these strengths and weaknesses in the next lab assignment when you tackle one of these frameworks yourself.
 
 ##Your First Client Templating
```
Comment (jagreene) [inline lessons/09-clientside-frameworks/README.md:33]: Do we still want to do this? Is client side templating right before learning frameworks actually useful?

File: lessons/09-clientside-frameworks/README.md [hunk 2]
```diff
@@ -1,51 +1,57 @@
...
+Luckily, there are really fantastic publically available and free frameworks you can utilize to organize your application.
+We will explore these in a bit.
 
 ##MVC on the client
-We talked last class about MVC and how **awesome** it is. 
-Hopefully you all have an appreciation for the nice structure and separation of concerns it affords you. 
-When we restructure our application for the client side, we now move that organizational structure to the client side, as well. 
-The direct implementation will differ based on the specific framework, but in general you will organize your **views** as templates that you load in only once when the first page loads. 
-The **models** are organized javascript objects with some additional functionality, such as a link to an API endpoint that can be used to retrieve and update data. 
+We talked last class about MVC and how **awesome** it is.
```
Comment (jagreene) [inline lessons/09-clientside-frameworks/README.md]: Did we? I don't see MVC mentioned in the last class's readme or did I hear us mention it in class

File: lessons/09-clientside-frameworks/README.md [hunk 3]
```diff
@@ -1,51 +1,57 @@
...
+Hopefully you all have an appreciation for the nice structure and separation of concerns it affords you.
+When we restructure our application for the client side, we now move that organizational structure to the client side, as well.
+The direct implementation will differ based on the specific framework, but in general you will organize your **views** as templates that you load in only once when the first page loads.
+The **models** are organized javascript objects with some additional functionality, such as a link to an API endpoint that can be used to retrieve and update data.
 The **controller** is also on the client where it handles rendering the templates using the appropriate model data.
 
 ##Popular Frameworks
-It turns out that there are a TON of different javascript client-side frameworks to choose from for any project. 
+It turns out that there are a TON of different javascript client-side frameworks to choose from for any project.
 These frameworks have various strengths and weaknesses. In this class we are going to explore AngularJS and ReactJS, however, you are more than welcome to explore other templates on your own.
```
Comment (jagreene) [inline lessons/09-clientside-frameworks/README.md]: Are we exploring both? I thought we decided on just doing React but I could be wrong

File: lessons/09-clientside-frameworks/README.md [hunk 4]
```diff
@@ -91,4 +97,92 @@ For lots of popular frameworks, there are often lots of people who have created
...
+- Which framework do you want to use?
+- How is what you're doing with a framework different from what you did with jQuery before?
+- How will the server communicate with the client?
+- On the board, map out the "boilerplate" for an app which uses a framework.
+
+
+<a name="homework-09"></a>
+### Before Class 10 (Tuesday 2/21/2017)
+#### Assignment
+Create a ToDo List App using a fronted framework, as for example shown [here](http://todomvc.com/examples/angularjs/#/), using Angular. Think about what routes your API needs to include for this application to function. Your app should be able to:
```
Comment (jagreene) [inline lessons/09-clientside-frameworks/README.md]: Create a ToDo List App using a frontend framework. [This](http://todomvc.com/examples/angularjs/#/) is a good example using Angular.

File: lessons/09-clientside-frameworks/README.md (hunk 1 above)
Comment (yunhsincynthiachen) [reply lessons/09-clientside-frameworks/README.md:33]: What do you mean?

File: lessons/09-clientside-frameworks/README.md (hunk 2 above)
Comment (yunhsincynthiachen) [reply lessons/09-clientside-frameworks/README.md]: Good catch. I'll add more about MVC

File: lessons/09-clientside-frameworks/README.md (hunk 1 above)
Comment (jagreene) [reply lessons/09-clientside-frameworks/README.md:33]: Sorry, highlighted the wrong line for the comment. I was referring to:

> So you are all very familiar with server side templating using handlebars.
>  -In the Lesson 9 folder there is an in_class folder containing the cat app that you are all familiar with. 		 +In the Lesson 9 folder there is an in_class folder containing the cat app that you are all familiar with.
>  -We are going to uncomment code to transform just one of the routes into an API endpoint and then use handlebars on the client side to render the content. 		 +We are going to uncomment code to transform just one of the routes into an API endpoint and then use handlebars on the client side to render the content.
>   		  
>  -You can probably imagine that organizing a very complex application with dozens of pages and models can get really complicated and messy really fast. 		 +You can probably imagine that organizing a very complex application with dozens of pages and models can get really complicated and messy really fast.

File: lessons/09-clientside-frameworks/README.md (hunk 3 above)
Comment (yunhsincynthiachen) [reply lessons/09-clientside-frameworks/README.md]: Yup! You're right just forget to change the phrasing there

**Review by billmwong — APPROVED** [review]

**Pull request merged**